features/GRD000_Grid-information.feature

```diff
@@ -0,0 +1,26 @@
+@implementer-agreement
+@GRD
+@POS
+@version1
+@E00020
+
+Feature: GRD000 - Grid Information
+    The rule verifies the presence of IFC entities used to define a design grid to be used as reference for object placement.
+    https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/concepts/Product_Shape/Product_Placement/Product_Grid_Placement/content.html
+
+
+    Scenario: Check for activation
+
+    Given an IfcGridPlacement
+    Given its attribute PlacesObject
+    Given its entity type is 'IfcProduct' including subtypes
+    Given return to IfcGridPlacement
+    Given its attribute PlacementRelTo
+    Given its attribute PlacesObject
+    Given its entity type is 'IfcGrid' 
+    Given return to IfcGridPlacement
+    Given its attribute PlacementLocation
+    Given IntersectingAxes = not empty
+
+    Then The IFC model contains information on quantities of elements
+
```

Comment:
So all in all, you could consider rewriting this scenario as:

```gherkin
Given an IfcProduct
And its attribute ObjectPlacement
And its entity type is IfcGridPlacement

Then The IFC model contains information on ...
```

Reply:
But would this be close enough to the [relating concept template](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/concepts/Product_Shape/Product_Placement/Product_Grid_Placement/content.html)? 
Would it be nice if we could state that some software supports a functional part based on these templates?

Reply:
Well, it goes back to the age old question that we have to second guess the intention behind the template, and this case I'd say it's twofold:

- [GRD000] Describe grid placement functionality: `IfcProduct ----ObjectPlacement---> GridPlacement`
- [GRD001-a] Constrain placement of grid *"a grid placement should be defined relatively to a grid"*: `GridPlacement ----PlacementRelTo----> IfcObjectPlacement ---PlacesObject---> IfcGrid`

And then a third component that is important, but is not in the template:

- [GRD001-b] *"The grid to which the placement is relative should contain the axes that are used for the intersection"*

![afbeelding](https://github.com/user-attachments/assets/0d78bcec-64e7-40bd-9c19-9171ad59f357)

Reply:
Let's phrase it differently, as soon as the model contains:

`IfcProduct ----ObjectPlacement---> GridPlacement`

The file contains a "grid placement" according to the functional part.

But if it's not according to the 2 further specifications I just highlighted, it's not that the model doesn't contain it, it's that the usage is invalid. So yes, these are additional normative parts and we need to create this separation.

Reply:
@evandroAlfieri @Ghesselink Annother rule for the backlog is this informal proposition from the docs:

> IntersectingAxes[1] and IntersectingAxes[2] shall not be part of the same row of grid axes, i.e. both shall not be within the same set of [IfcGrid](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcGrid.htm).UAxes or [IfcGrid](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcGrid.htm).VAxes of the corresponding [IfcGrid](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcGrid.htm).

https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcVirtualGridIntersection.htm